core/trino-main/src/test/java/io/trino/sql/planner/TestConnectorExpressionTranslator.java

```diff
@@ -177,6 +180,41 @@ public void testTranslateComparisonExpression(ComparisonExpression.Operator oper
                 .collect(toDataProvider());
     }
 
+    @Test(dataProvider = "testTranslateArithmeticBinaryDataProvider")
+    public void testTranslateArithmeticBinary(ArithmeticBinaryExpression.Operator operator)
```

Comment:
the tests don't make it clear that data types might differ of the two arguments. e.g. `TINYINT '1' + INT '1'` or even `INT '1' + DOUBLE '1.23'` are both valid expressions.

Reply:
They cannot differ. Addition of values of two different types involves a coercion. 
Casts to be handled later: https://github.com/trinodb/trino/pull/11522

Reply:
I see no verification in the implementation. Seems we consider that the engine will have done it's job already?

Reply:
@hashhar what would you want to see verified?

Reply:
That the type matches of lhs and rhs in `visitArithmeticBinary`?

But I see that it doesn't help a lot.